src/workerd/io/io-context.c++

```diff
@@ -1089,6 +1089,16 @@ void IoContext::runImpl(Runnable& runnable,
         // Check if we hit a limit.
         limitEnforcer->requireLimitsNotExceeded();
 
+        // If we were terminated because abort() was called, then it's not an unknown
+        // reason...
+        if (!abortFulfiller->isWaiting()) {
+          // Nothing to do here. Do not log anything. The assumption is that we've
+          // terminated because the IoContext was aborted and isolate->TerminateExection()
+          // was called (likely because of someone using process.exit(...) in Node.js
+          // compat mode).
+          return;
```

Comment:
You can't just return here. The caller may assume that this means the call actually completed successfully. This could, for example, cause an alarm handler to think that the alarm finished successfully and therefore doesn't need to be retried. This could lead to data loss.

You can either:
a. Return `abortPromise.addBranch()`, to propagate the abort exception.
b. Return `kj::NEVER_DONE` and count on the caller joining with the abort exception.

I think (a) is better.